sdk-setup/src/sdk-manage

```diff
@@ -1565,13 +1565,23 @@ target_snapshot_needs_reset() {
         return 1
     fi
 
-    # TODO how to determine recent package removal?
     local most_recent_install_time=
     most_recent_install_time=$(set -o pipefail; rpm --root="$MER_TARGETS/$original" -qa \
-        --queryformat '%{installtime}\n' |sort -n |tail -n1) || return
-    local snapshot_time_s=$(date --date "$snapshot_time" +%s) || return
+        --queryformat '%{installtime}\n' | sort -n | tail -n1) || return
+
+    # The rpm database does not know about the most recent package removal, so
+    # parse the zypper install log. zypp/history looks like this:
+    # 2019-02-27 16:39:18|remove |droid-hal-mydevice-tools|0.0.6-201902271629|armv7hl|user@pc|
+    local most_recent_uninstall_time=$(set -o pipefail; \
+        tac $MER_TARGETS/$original/var/log/zypp/history \
+        |  grep -m 1 "|remove" \
+        | cut -d "|" -f 1 \
```

Comment:
This would also match `2019-02-27 16:39:18|install |remove-foo|1.2.3|armv7hl|user@pc|`

Consider something line `sed -n 's/^\([^|]\+\)|remove.*/\1/p'` instead of grep+cut here.

Reply:
Used this format instead(note the additional `cut`):
```
tac ... \
|cut -d "|" -f "1,2" \
|grep -m 1 "|remove" \
|cut -d "|" -f 1 \
```
I've timed 100 runs against your `sed` method and it's a tad faster, 1.3s(non-sed) vs 1.6s(sed).

Reply:
ok :)